Sources/Fuzzilli/Core/MutationEngine.swift

```diff
@@ -153,69 +136,23 @@ public class MutationEngine: ComponentBase, FuzzEngine {
             }
-            
+
             if !mutated {
                 logger.warning("Could not mutate sample, giving up. Sample:\n\(fuzzer.lifter.lift(parent))")
                 continue
             }
-    
-            fuzzer.dispatchEvent(fuzzer.events.ProgramGenerated, data: program)
-            
-            let execution = fuzzer.execute(program)
-            
-            switch execution.outcome {
-            case .crashed(let termsig):
-                // For crashes, we append a comment containing the content of stderr
-                program = appendComment("Stderr:\n" + execution.stderr, to: program)
-                fuzzer.processCrash(program, withSignal: termsig, origin: .local)
-                
-            case .succeeded:
-                mutator.producedValidSample()
-                fuzzer.dispatchEvent(fuzzer.events.ValidProgramFound, data: program)
-                
-                if let aspects = fuzzer.evaluator.evaluate(execution) {
-                    fuzzer.processInteresting(program, havingAspects: aspects, origin: .local)
-                    // Continue mutating the parent as the new program should be in the corpus now.
-                    // Moreover, the new program could be empty due to minimization, which would cause problems above.
-                } else {
-                    // Continue mutating this sample
-                    parent = program
-                }
-                
-            case .failed:
-                mutator.producedInvalidSample()
-                if fuzzer.config.diagnostics {
-                    program = appendComment("Stdout:\n" + execution.stdout, to: program)
-                }
-                fuzzer.dispatchEvent(fuzzer.events.InvalidProgramFound, data: program)
-                
-            case .timedOut:
-                mutator.producedInvalidSample()
-                fuzzer.dispatchEvent(fuzzer.events.TimeOutFound, data: program)
-            }
-        }
-    }
-    
-    private func appendComment(_ comment: String, to program: Program) -> Program {
-        var code = program.code
-        code.append(Instruction(Comment(comment)))
-        return Program(with: code)
-    }
-    
-    private func makePrefix() -> Program {
-        let b = fuzzer.makeBuilder()
-        
-        for generator in programPrefixGenerators {
-            b.run(generator)
-        }
 
-        let prefixProgram = b.finalize()
+            let (outcome, newCoverage) = self.execute(program, stats: &mutator.stats)
 
-        fuzzer.collectRuntimeTypes(for: prefixProgram)
-        
-        return prefixProgram
+            // If we have new coverage continue mutating the parent as the new program should be in the corpus now.
+            // Moreover, the new program could be empty due to minimization, which would cause problems above.
+            // If we don't have new coverage, we continue mutating this sample.
+            if .succeeded == outcome && !newCoverage {
```

Comment:
This must be a pretty old comment :D I think we can just remove the `&& !newCoverage` bit and just always continue mutating the program if the outcome is `.succeeded`. I think then you can also remove the 2nd return value of `execute`, then it's simpler